lib/cartopy/tests/mpl/test_mpl_integration.py

```diff
@@ -506,6 +506,37 @@ def test_pcolormesh_set_array_with_mask(mesh_data_kind):
     return fig
 
 
+def test_pcolormesh_set_array_nowrap():
+    # Roundtrip check that set_array works with the correct shaped arrays
+    nx, ny = 36, 18
+    xbnds = np.linspace(-60, 60, nx, endpoint=True)
+    ybnds = np.linspace(-80, 80, ny, endpoint=True)
+    xbnds, ybnds = np.meshgrid(xbnds, ybnds)
+
+    rng = np.random.default_rng()
+    print(f"{xbnds.shape=}, {ybnds.shape=}")
+    data = rng.random((17, 35))
```

Comment:
Here and later, I think it's better to use `ny - 1` and `nx - 1`.

Reply:
Done